enterprise/decorators.py

```diff
@@ -134,11 +140,15 @@ def wrapper(request, *args, **kwargs):
         # Enterprise-linked IdP and the pipeline will get them back here.
         if not request.user.is_authenticated():
             parsed_current_url = urlparse(request.get_full_path())
-            parsed_query_string = parse_qs(parsed_current_url.query)
-            parsed_query_string.update({
+            parsed_query_string = parse_qsl(parsed_current_url.query)
```

Comment:
How does parsing this to a list of key/value pairs instead of a dictionary solve the problem?

Reply:
@douglashall previously the `parse_qs` was converting the values of all the existing query params to list so the `catalog=6eca3efb-f3a0-4c08-806f-c6e6b65d61cb` would become `{'catalog': ['6eca3efb-f3a0-4c08-806f-c6e6b65d61cb'] }`. `parse_qs` does this to combine the values of same params.

`parse_qsl` on the other hand converts the query params to tuples this way the query param values remain unchanged.

https://stackoverflow.com/questions/1024143/how-to-stop-python-parse-qs-from-parsing-single-values-into-lists

Reply:
@zubair-arbi Got it. Makes sense. Thanks.

Reply:
Actually, no. There's simpler fix here. See below.

Reply:
@douglashall I intentionally didn't used that approach as it says ` Please note that if there are multiple values for the same key, you'll only get the last one.` I know we aren't currently using multiple values for the same key but I wanted to implement a flexible solution. Please let me know your thoughts.

Reply:
@zubair-arbi That comment you are referencing from the stackoverflow article you linked to has nothing to do with the `doseq` approach.